integrations/watsonx/src/haystack_integrations/components/generators/watsonx/chat/chat_generator.py

```diff
@@ -0,0 +1,391 @@
+            if content:
+                chunk_meta = {
+                    "model": self.model,
+                    "index": chunk["choices"][0].get("index", 0),
+                    "finish_reason": chunk["choices"][0].get("finish_reason"),
+                    "received_at": datetime.now(timezone.utc).isoformat(),
+                }
+                streaming_chunk = StreamingChunk(content=content, meta=chunk_meta)
+                chunks.append(streaming_chunk)
+                callback(streaming_chunk)
+
+        return {"replies": [self._convert_streaming_chunks_to_chat_message(chunks)]}
+
+    def _handle_standard(self, api_args: dict[str, Any]) -> dict[str, list[ChatMessage]]:
+        """Handle synchronous standard response."""
+        response = self.client.chat(messages=api_args["messages"], params=api_args["params"])
+        return self._process_response(response)
+
+    async def _handle_async_streaming(
+        self,
+        api_args: dict[str, Any],
+        callback: StreamingCallbackT,
+    ) -> dict[str, list[ChatMessage]]:
+        """Handle asynchronous streaming response."""
+        chunks: list[StreamingChunk] = []
+        stream_generator = await self.client.achat_stream(messages=api_args["messages"], params=api_args["params"])
+
+        async for chunk in stream_generator:
+            if not isinstance(chunk, dict) or not chunk.get("choices"):
+                continue
+
+            content = chunk["choices"][0].get("delta", {}).get("content", "")
+            if content:
+                chunk_meta = {
+                    "model": self.model,
+                    "index": chunk["choices"][0].get("index", 0),
+                    "finish_reason": chunk["choices"][0].get("finish_reason"),
+                    "received_at": datetime.now(timezone.utc).isoformat(),
+                }
+                streaming_chunk = StreamingChunk(content=content, meta=chunk_meta)
+                chunks.append(streaming_chunk)
+                try:
+                    result = callback(streaming_chunk)
+                    if asyncio.iscoroutine(result):
+                        await result
+                except Exception:
+                    # Fallback to sync call if async fails
+                    callback(streaming_chunk)
+        return {"replies": [self._convert_streaming_chunks_to_chat_message(chunks)]}
+
+    def _convert_streaming_chunks_to_chat_message(self, chunks: list[StreamingChunk]) -> ChatMessage:
+        """Convert list of streaming chunks to a single ChatMessage."""
+        if not chunks:
+            return ChatMessage.from_assistant("")
+
+        content = "".join(chunk.content for chunk in chunks)
+        last_chunk_meta = chunks[-1].meta if chunks else {}
+
+        return ChatMessage.from_assistant(
+            text=content,
+            meta={
+                "model": self.model,
+                "finish_reason": last_chunk_meta.get("finish_reason"),
+                "usage": {},
```

Comment:
Do you happen to know if WatsonX never returns usage or if it's just easier not to support currently?

If the latter let me know and I can open an issue for it to add a later time.